### ISSUE TITLE
old fashioned validation does not work after registry change

### ISSUE THREAD

**Issue by mosheliv** [open-mmlab/mmdetection#997]
Thank you for the impressive framework!
I have used https://github.com/open-mmlab/mmdetection/pull/909 before, but it uses an api that is now gone, get dataset.
Is there a way of getting the datasets now? this registry abstraction is confusing.
Also, it would be good (and save you lots of issues) if you can explain how you validation hooks works. The only way I know of to see when the model starts overfitting is by looking at the val_loss and val_acc. I just can't see these in your output.

**Comment by hellock** [comment]
The old `get_dataset` is equivilent to [`build_dataset`](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/datasets/builder.py#L30) in the latest version.

**Comment by mosheliv** [comment]
thanks, just figured it out myself. It would be great if this will also be available without modifying mmdetection code.

**Comment by hellock** [comment]
Wouldn't it be better if you just use the evaluation hook to test the val set with `--validate`?

**Comment by mosheliv** [comment]
It probably would, but I have a single gpu (well, I actually have 2 but one is busy doing something else). So, no distributed -> no validation.....
Most people in the world dealing with Data Science have only one GPU. 
It is also not explained anywhere how do you actually do that so I am not certain what it is.

**Comment by hellock** [comment]
I see. We will have better support for single GPU training.

**Comment by mosheliv** [comment]
Thank you so much for maintaining this excellent repository. I will try to help but as you are restructuring things, I don't think it will do much good.

**Comment by mosheliv** [comment]
Sorry to reopen this, but it seems the method described actually train on the validation set, so it is being useless. Is there a way to prevent mmdetection from training on the val set while calculating loss and accuracy?

**Comment by hellock** [comment]
May be related to #1058

**Comment by mosheliv** [comment]
This only works in distributed. I am using single gpu
